src/ClientData/ScopeIdProvider.cpp

```diff
@@ -32,14 +32,18 @@ std::unique_ptr<NameEqualityScopeIdProvider> NameEqualityScopeIdProvider::Create
                 ->function_id();
 
   absl::flat_hash_map<uint64_t, const ScopeInfo> scope_id_to_info;
+  absl::flat_hash_map<const ScopeInfo, uint64_t> scope_info_to_id;
```

Comment:
are we using both of them?

Reply:
Yes, we do. `IdProvider` does two things. 
1. Map timer_info to `scope_id`
2. Map `scope_id` to `scope_info`.